### PR TITLE
Emotional expressivity update

### PR DESCRIPTION
**Pull request by kcmcveigh**

### CODE REVIEW

File: openwillis/measures/video/facial_emotion.py [hunk 1]
```diff
@@ -5,19 +5,159 @@
...
+    emos_aus = np.hstack([emotions,aus])
+    df_emo = pd.DataFrame([emos_aus],columns=emo_cols)
+    df_emo['mouth_openness'] = mouth_openness(landmarks)
+
+    return df_emo
+
+def run_pyfeat(path, skip_frames=5, bbox_list=[]):
     """
     ------------------------------------------------------------------------------------------------------
     This function takes an image path and measures config object as input, and uses the DeepFace package to
```
Comment (GeorgeEfstathiadis) [inline openwillis/measures/video/facial_emotion.py:163]: Replace DeepFace with py-feat?

File: openwillis/measures/video/facial_emotion.py [hunk 2]
```diff
@@ -39,50 +179,75 @@ def run_deepface(path, measures):
...
-            logger.info(f'Face not detected by deepface in : {path}')
+            logger.info(f'Face not detected by pyfeat in : {path}')
 
     return df_list
 
-def get_undected_emotion(frame, cols):
+def get_undected_emotion(frame, cols, fps):
     """
     ------------------------------------------------------------------------------------------------------
     This function returns a pandas dataframe with a single row of NaN values for the different facial emotion
```
Comment (GeorgeEfstathiadis) [inline openwillis/measures/video/facial_emotion.py:253]: Replace in description DeepFace with py-feat?

File: openwillis/measures/video/facial_emotion.py [hunk 3]
```diff
@@ -5,19 +5,159 @@
...
+def detect_emotions(detector, frame, emo_cols, bb_dict={},threshold=.95):
+
+    faces = get_faces(
+        detector,
+        frame,
+        bb_dict,
+        threshold=threshold
+    )
+
+    if len(faces[0])<1:
```
Comment (GeorgeEfstathiadis) [inline openwillis/measures/video/facial_emotion.py:134]: Add check for frames that didn't have a bounding box

```suggestion
    if len(faces[0])<1 or np.isnan(faces[0][0][:4]).sum() == 4:
```

File: openwillis/measures/video/facial_emotion.py (hunk 1 above)
Comment (kcmcveigh) [reply openwillis/measures/video/facial_emotion.py:175]: Yep! This is just a q right? No suggestion?

**Review by GeorgeEfstathiadis — APPROVED** [review]

**Pull request merged**